projects/iio_demo_freeRTOS/src/platform/maxim/main.c

```diff
@@ -0,0 +1,151 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "platform_includes.h"
+#include "common_data.h"
+
+#ifdef IIO_EXAMPLE
+#include "iio_example.h"
+#endif
+
+/**
+ * @brief IIOD example task
+ * @return Returns the error code of iio_example_main function
+*/
+int iiodTask()
+{
+	return iio_example_main();
+}
+
+/**
+ * @brief LED blinking task for multithreading example
+ * @return Returns error code
+*/
+int blinkingTask()
+{
+	int ret = 0;
+	struct no_os_gpio_desc *led_pin;
+	ret = no_os_gpio_get(&led_pin, &gpio_ip);
+	if (ret)
+		goto error_pin;
+
+	ret = no_os_gpio_direction_output(led_pin, NO_OS_GPIO_LOW);
+	if (ret)
+		goto error_pin;
+
+	bool led_on = true;
+	while (1) {
+		if (!led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_HIGH);
+			if (ret)
+				goto error_pin;
+			led_on=true;
+			no_os_mdelay(100);
+		}
+		if (led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_LOW);
+			if (ret)
+				goto error_pin;
+			led_on=false;
+			no_os_mdelay(100);
+		}
+	}
+
+error_pin:
+	no_os_gpio_remove(led_pin);
+	return ret;
+}
+
+/**
+ * @brief Creating iio example with blinking led task
+ * @return  Returns error code
+ *
+*/
+int create_tasks(void)
+{
+	int ret;
+	TaskHandle_t iio_app_handle = NULL;
+	TaskHandle_t led_task_handle = NULL;
+
+	ret = xTaskCreate(iiodTask, (const char *)"iiodTask", configIIO_APP_STACK_SIZE,
+			  NULL, tskIDLE_PRIORITY + 1, &iio_app_handle);
+	if (ret != pdPASS) {
+		printf("xTaskCreate() failed to create iiodTask task.\n");
+		if (iio_app_handle != NULL)
+			vTaskDelete(iio_app_handle);
+		return ret;
+	}
+	ret = xTaskCreate(blinkingTask, (const char *)"blinkingTask",
+			  configMINIMAL_STACK_SIZE, NULL, tskIDLE_PRIORITY + 1, &led_task_handle);
+	if (ret != pdPASS) {
+		printf("xTaskCreate() failed to create blinkingTask task.\n");
+		if (led_task_handle != NULL)
+			vTaskDelete(led_task_handle);
+		return ret;
```

Comment:
in case iiodTask is created successfully, shouldn't we also delete that task before returning?

Reply:
In this case I think it is more up to the developer. Maybe one task fails, but you might still want another driver to publish some data. (but in this case I still have to remove the returns :) )

Reply:
Also in case someone links the FreeRTOS_CLI for terminal access, you can't stop the entire rtos, just because one task is failing.

Reply:
As discussed will delete all the tasks in case of an error.